source/spirv_validator_options.h

```diff
@@ -43,12 +43,14 @@ struct spv_validator_options_t {
         relax_struct_store(false),
         relax_logical_pointer(false),
         relax_block_layout(false),
+        scalar_block_layout(false),
         skip_block_layout(false) {}
 
   validator_universal_limits_t universal_limits_;
   bool relax_struct_store;
   bool relax_logical_pointer;
   bool relax_block_layout;
+  bool scalar_block_layout;
```

Comment:
What do you think about turning this and relax_block_layout in a Layout enum? The setters could stay as is, but make it so if relaxed block layout won't downgrade from scalar block layout. I think it might make the checking code easier to read since you'd only have to compare against the one enum value instead of combining multiple boolean values.

Reply:
I took a look at doing this.  The trouble is that the current public API allows you to set each of these indivdal bits to _false_.  So if I've set   relax=true, then scalar=true, then scalar=false, then I need extra bits to store the fact that the final result should be relax=true instead of something else.